flink-test-utils-parent/flink-connector-test-utils/src/main/java/org/apache/flink/connector/testutils/source/reader/SourceReaderTestBase.java

```diff
@@ -136,12 +145,12 @@ public void testSnapshot() throws Exception {
         ValidatingSourceOutput output = new ValidatingSourceOutput();
         // Add a split to start the fetcher.
         List<SplitT> splits =
-                getSplits(NUM_SPLITS, NUM_RECORDS_PER_SPLIT, Boundedness.CONTINUOUS_UNBOUNDED);
+                getSplits(numSplits, NUM_RECORDS_PER_SPLIT, Boundedness.CONTINUOUS_UNBOUNDED);
         try (SourceReader<Integer, SplitT> reader =
-                consumeRecords(splits, output, NUM_SPLITS * NUM_RECORDS_PER_SPLIT)) {
```

Comment:
I prefer to use the `getTotalNumRecords` method to replace all the logic `numSplits * NUM_RECORDS_PER_SPLIT` where means to get the total number of records.

Reply:
I have turned it into a variable `totalNumRecords` and changed here.